Report/Chapters/Design.tex

```diff
@@ -1,17 +1,103 @@
 \chapter{Design}
 
+This chapter contains a high level description of the design choices made in the process of creating Viskell.
+The specific implementation details are discussed in chapter \ref{chap:implementation} on page \pageref{chap:implementation}.
+
+\section{User interface}
+
+The user interface basically consists of two parts, namely the menus for the different options a user can select and the part where the user can manipulate functions and create their own (the canvas).
+
+\subsection{Graphical design}
+
+\subsubsection{Blocks and lines}
+
+We chose to represent Haskell functions by function blocks since this is a logical element for users, a combination of blocks forms a program.
+Each block consists of a function name and its associated arguments. With each argument comes an anchor node to which other block can be connected through lines.
+
+In Haskell, values are being passed to functions without being saved to memory.
+The lines between blocks represent this flow.
+Thus, a program can be built like an electronic circuit.
+
+We chose to make a distinction between three kinds of blocks.
+The first kind in an input block, which represents a value.
+This input block can be connected to other blocks to use the value as an input.
+Besides an input block that takes a string as input and automatically detects the type, there is also a slider available.
+
+The second kind of block is a function block. This block has a number of inputs and a single output, and defines the number of inputs that are going to be used.
+The number of inputs can be changed by moving a knot over the function arguments.
+This allows for passing functions as input for other functions.
```

Comment:
I don't know if this is partial application or currying, but you could mention one of those here.